builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1368,6 +1565,7 @@ func (vm *virtualMachine) createVirtualMachine(c *govmomi.Client) error {
 	for _, hd := range vm.hardDisks {
```

Comment:
In order to keep track of hard disk state that can later be used to deal with upgrade, we need to persist two pieces of information.  The first is the path to the VMDK (required to know if disk is renamed, moved, etc.) and the second being the device name/id for the disk mounted to a vm (required to detach disk from VM at a later point).